File: COM_backend/src/main/java/com/uos/comp6239backend/service/THistoryService.java
package com.uos.comp6239backend.service;

import com.uos.comp6239backend.tdata.entity.ReaderStoryHistory;
import com.uos.comp6239backend.tdata.entity.ReadingPath;
import com.uos.comp6239backend.tdata.entity.ReadingPathItem;
import com.uos.comp6239backend.utils.ResponseMap;

import java.util.List;
import java.util.Map;

/**
 * @title: THistoryService
 * @Author Hym
 * @Date: 2024-05-12 21:03
 * @Description:
 * @Version 1.0
 */
public interface THistoryService {
    //以下是history表
    //添加新的阅读历史记录
    ResponseMap.ResultData insertReaderHistory(Map<String, Object> values);

    //更新现有的阅读历史，例如更新笔记或当前进度。
    ResponseMap.ResultData updateReaderHistory(Map<String, Object> values);

    //根据读者ID查找其所有的阅读历史
    ResponseMap.ResultData findHistoryByReaderId(Map<String, Object> values);

    //根据故事ID查找所有相关的阅读历史
    ResponseMap.ResultData findHistoryByStoryId(Map<String, Object> values);

    //t_reading_path
//    创建新的阅读路径
    ResponseMap.ResultData createReadingPath(Map<String, Object> values);

    //    更新阅读路径，如修改起始阅读路径项
    ResponseMap.ResultData updateReadingPath(Map<String, Object> values);

    //    根据路径ID获取阅读路径详情。
    ResponseMap.ResultData getReadingPathById(Map<String, Object> values);

    //    根据故事ID获取所有相关的阅读路径。
    ResponseMap.ResultData getPathsByStoryId(Map<String, Object> values);

    //以下是item表
//    添加新的阅读路径项。
    ResponseMap.ResultData insertPathItem(Map<String, Object> values);

    //    更新现有的阅读路径项，如阅读次数。
    ResponseMap.ResultData updatePathItem(Map<String, Object> values);

    //    根据章节ID获取所有路径项
    ResponseMap.ResultData getPathItemsByChapterId(Map<String, Object> values);

    //    根据路径项ID获取详情
    ResponseMap.ResultData getPathItemById(Map<String, Object> values);
}

File: COM_project/app/src/main/java/com/codelab/basiclayouts/data/service/THistoryService.kt
package com.codelab.basiclayouts.data.service

import com.codelab.basiclayouts.data.ResultData
import com.codelab.basiclayouts.model.reader.readerStoryHistory
import com.codelab.basiclayouts.model.reader.readingPath
import com.codelab.basiclayouts.model.reader.readingPathItem
import retrofit2.http.Body
import retrofit2.http.POST

interface THistoryService {

    @POST("/THistoryCtrl/insertReaderHistory")
    suspend fun insertReaderHistory(@Body values: readerStoryHistory): ResultData<readerStoryHistory> // 无返回类型

    @POST("/THistoryCtrl/updateReaderHistory")
    suspend fun updateReaderHistory(@Body values: readerStoryHistory): ResultData<readerStoryHistory>

    @POST("/THistoryCtrl/findHistoryByReaderId")
    suspend fun findHistoryByReaderId(@Body values: Map<String, Int>): ResultData<List<readerStoryHistory>>

    @POST("/THistoryCtrl/findHistoryByStoryId")
    suspend fun findHistoryByStoryId(@Body values: Map<String, Int>): ResultData<List<readerStoryHistory>>

    @POST("/THistoryCtrl/createReadingPath")
    suspend fun createReadingPath(@Body values: readingPath): ResultData<Int>

    @POST("/THistoryCtrl/updateReadingPath")
    suspend fun updateReadingPath(@Body values: readingPath): ResultData<readingPath>

    @POST("/THistoryCtrl/getReadingPathById")
    suspend fun getReadingPathById(@Body values: Map<String, Int>): ResultData<readingPath>

    @POST("/THistoryCtrl/getPathsByStoryId")
    suspend fun getPathsByStoryId(@Body values: Map<String, Int>): ResultData<List<readingPath>>

    @POST("/THistoryCtrl/insertPathItem")
    suspend fun insertPathItem(@Body values: readingPathItem): ResultData<Int>

    @POST("/THistoryCtrl/updatePathItem")
    suspend fun updatePathItem(@Body values: readingPathItem): ResultData<readingPathItem>

    @POST("/THistoryCtrl/getPathItemsByChapterId")
    suspend fun getPathItemsByChapterId(@Body values: Map<String, Int>): ResultData<List<readingPathItem>>

    @POST("/THistoryCtrl/getPathItemById")
    suspend fun getPathItemById(@Body values: Map<String, Int>): ResultData<readingPathItem>
}